src/openforms/products/utils/tests/test_string.py

```diff
@@ -0,0 +1,59 @@
+from django.test import TestCase
+
+from ..entry import convert_json_schema_to_py
+
+
+class SingleTextFieldTestCase(TestCase):
+    def setUp(self):
+        self.json_schema_string = {
+            "properties": {
+                "comment": {
+                    "title": "Comment",
+                    "type": "string",
+                    "description": "Few words in short",
+                    "maxLength": 250,
+                }
+            },
+        }
+        self.json_schema_string_pattern = {
+            "properties": {
+                "kenteken": {
+                    "title": "Kenteken",
+                    "type": "string",
+                    "pattern": "^[A-Z0-9]?$",
+                    "description": "Aanvraag wordt gedaan.",
+                    "maxLength": 250,
+                }
+            },
+        }
+
+        self.json_schema_textarea = {
+            "required": [],
+            "properties": {
+                "review": {
+                    "title": "User review",
+                    "type": "string",
+                    "description": "Product review",
+                }
+            },
+        }
+
+    def test_string(self):
+        result = convert_json_schema_to_py(self.json_schema_string)
+        self.assertEqual(result["components"][0]["type"], "textfield")
+        self.assertEqual(result["components"][0]["key"], "comment")
+        self.assertEqual(result["components"][0]["description"], "Few words in short")
+
+    def test_string_with_pattern(self):
+        result = convert_json_schema_to_py(self.json_schema_string_pattern)
+        self.assertEqual(result["components"][0]["type"], "textfield")
+        self.assertEqual(
+            result["components"][0]["validate"].get("pattern"), "^[A-Z0-9]?$"
+        )
+
+    def test_string_textarea(self):
+        result = convert_json_schema_to_py(self.json_schema_textarea)
+        self.assertEqual(result["components"][0]["type"], "textarea")
+        self.assertEqual(result["components"][0]["key"], "review")
+        self.assertEqual(result["components"][0]["description"], "Product review")
+        self.assertEqual(result["components"][0]["validate"].get("required"), False)
```

Comment:
Please keep your test data setup, test call and assertions together in the test method. Now the `setUp` method sets up the test data for ALL the tests, but there's only ever one thing relevant.

The most readable test cases don't have a `setUp` method and follow the pattern:

```
def test_something(...):
    # set up test data, one or multiple lines

    # call the thing you're testing

    # make assertions, one or more lines
```

Note the deliberate blank lines to separate the set up / test / assertions parts of a test. Ideally, a test is understood by looking at just the method and nothing more.